### PR TITLE
Ssh clean up

### PR DESCRIPTION
**Pull request by jayjb**
## Proposed changes

This commit introduces a few changes:
- ssh key paths if you would like to generate your own and supply the path
- ssh code clean up
- config code cleanup enabling checks for ports and config variables. 

## Types of changes

What types of changes does your code introduce to this repository?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation Update

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] Lint and unit tests pass locally with my changes (if applicable)
- [x] I have run pre-commit (`pre-commit` in the repo)
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: opencanary/config.py [hunk 1]
```diff
@@ -93,52 +99,19 @@ def getVal(self, key, default=None):
...
-                self.valid(key, value)
+                self.is_valid(key, value)
             except ConfigException as e:
                 errors.append(e)
 
         # Test that no ports overlap
-        ports = {k: v for k, v in iteritems(self.__config) if k.endswith(".port")}
-        newports = {k: v for k, v in iteritems(params) if k.endswith(".port")}
-        ports.update(newports)
+        ports = {k: int(v) for k, v in iteritems(params) if k.endswith(".port")}
```
Comment (benjamin-thinkst) [inline opencanary/config.py:114]: Might as well drop `iteriterms` from `six` and use `params.items()`

File: opencanary/config.py [hunk 2]
```diff
@@ -43,6 +44,11 @@ def detectIPTables():
         return False
 
 
+SERVICE_REGEXES = {
+    "ssh.version": r"SSH-(2.0|1.5|1.99|1.0)-([!-,\-./0-~]+(:?$|\s))(?:[ -~]*)$",
```
Comment (benjamin-thinkst) [inline opencanary/config.py:48]: Do we want to add the length constraint here too.
see https://github.com/thinkst/opencanary/pull/309/files#diff-5f3e79391c8c84f0500a2fbfaab887c2661e9e86cb9f26cdeb9cb6da69a0a308L194

**Review by benjamin-thinkst — APPROVED** [review]
Looks good. Some suggestions that may suit this change.

File: opencanary/config.py (hunk 1 above)
Comment (jayjb) [reply opencanary/config.py:114]: thanks done!

File: opencanary/config.py (hunk 2 above)
Comment (jayjb) [reply opencanary/config.py:48]:
```
        if key == "ssh.version" and len(val) > 253:
            raise ConfigException(key, "SSH version string too long (%s..)" % val[:5])
```
We already have this which results in a nicer error message too. So going to leave it as is

**Pull request merged**